pkg/provider/garden/ruleset/securityhardenedshoot/rules/2000_test.go

```diff
@@ -0,0 +1,102 @@
+// SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package rules_test
+
+import (
+	"context"
+
+	gardencorev1beta1 "github.com/gardener/gardener/pkg/apis/core/v1beta1"
+	"github.com/gardener/gardener/pkg/client/kubernetes"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/ptr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	fakeclient "sigs.k8s.io/controller-runtime/pkg/client/fake"
+
+	"github.com/gardener/diki/pkg/provider/garden/ruleset/securityhardenedshoot/rules"
+	"github.com/gardener/diki/pkg/rule"
+)
+
+var _ = Describe("#2000", func() {
+	var (
+		fakeClient     client.Client
+		ctx            = context.TODO()
+		shootName      = "foo"
+		shootNamespace = "bar"
+
+		shoot    *gardencorev1beta1.Shoot
+		r        rule.Rule
+		ruleName = "Shoot clusters must have anonymous authentication disabled for the Kubernetes API server."
+		ruleID   = "2000"
+		severity = rule.SeverityHigh
+	)
+
+	BeforeEach(func() {
+		fakeClient = fakeclient.NewClientBuilder().WithScheme(kubernetes.GardenScheme).Build()
+
+		shoot = &gardencorev1beta1.Shoot{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      shootName,
+				Namespace: shootNamespace,
+			},
+		}
+
+		r = &rules.Rule2000{
+			Client:         fakeClient,
+			ShootName:      shootName,
+			ShootNamespace: shootNamespace,
+		}
+	})
+
+	DescribeTable("Run cases", func(updateFn func(), expectedCheckResults []rule.CheckResult) {
```

Comment:
```suggestion
	DescribeTable("Run cases", func(updateFn func(), expectedCheckResult rule.CheckResult) {
```
Last comment, no need for `expectedCheckResults` to be a slice when the function can only return a single `rule.CheckResult`